src/vsr/client.zig

```diff
@@ -568,9 +797,11 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
             self.message_bus.send_message_to_replica(replica, message);
         }
 
-        fn send_request_for_the_first_time(self: *Self, message: *Message) void {
-            assert(self.request_queue.head_ptr().?.message == message);
+        fn send_request_for_the_first_time(self: *Self) void {
```

Comment:
Need to avoid batching into the head Request of the queue if it's currently inflight (message has already been sent to MessageBus). The implicit contract is indeed kind of a hack.